### PR TITLE
feat: Add label with results of filtering in select and multiselect

### PR DESCRIPTION
**Pull request by Al-Dani**
### Description

Adds new property to select, multiselect and autosuggest components to specify number of matches while filtering.
Resulted text is displayed in status field of dropdown.

Related links, issue #, if available: n/a

### How has this been tested?

New unit test added.
Try in 
* [select.sync with auto filtering](https://d21d5uik3ws71m.cloudfront.net/components/7c23f4ac91871261a188e9ffddce994f18455889/index.html#/light/select/select.sync.example)
* [select.async](https://d21d5uik3ws71m.cloudfront.net/components/7c23f4ac91871261a188e9ffddce994f18455889/index.html#/light/select/select.async.example)
<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/property-filter/property-filter-autosuggest.tsx [hunk 1]
```diff
@@ -34,7 +34,9 @@ import { joinStrings } from '../internal/utils/strings';
 const DROPDOWN_WIDTH_OPTIONS_LIST = 300;
 const DROPDOWN_WIDTH_CUSTOM_FORM = 200;
 
-export interface PropertyFilterAutosuggestProps extends AutosuggestProps, InternalBaseComponentProps {
+export interface PropertyFilterAutosuggestProps
+  extends Omit<AutosuggestProps, 'filteringResultsText'>,
```
Comment (timogasda) [inline src/property-filter/property-filter-autosuggest.tsx:38]: So we don't want to implement this for Property filter? Why?

File: src/select/__tests__/select.test.tsx [hunk 2]
```diff
@@ -305,6 +305,28 @@ describe.each([false, true])('expandToViewport=%s', expandToViewport => {
     });
   });
 
+  test('shows filtering result text after typing', () => {
```
Comment (timogasda) [inline src/select/__tests__/select.test.tsx:308]: Would be nice to have a few more tests covering how the filtering results text "competes" with the other types of text that can be displayed in the dropdown. Maybe this is something for the `dropdown-status` tests.

**Review by timogasda — CHANGES_REQUESTED** [review]

File: src/property-filter/property-filter-autosuggest.tsx (hunk 1 above)
Comment (Al-Dani) [reply src/property-filter/property-filter-autosuggest.tsx:38]: We decided that Property Filter and Attribute editor will be out of scope. We didn't receive requests for them, so if there will be some in the future, we can do a follow-up.

One more thing is that Property filter needs more investigation from a design perspective and it's more tricky to come up with meaningful writing guidelines

Comment (timogasda) [reply src/property-filter/property-filter-autosuggest.tsx:38]: Cool, works for me!

**Pull request merged**